src/core/nixl_agent.cpp

```
@@ -64,11 +64,13 @@ nixlXferReqH::nixlXferReqH(const std::string &remote_agent,
                            const nixl_mem_t local_type,
                            const nixl_mem_t remote_type,
                            const size_t desc_count,
-                           const nixl_remote_section_weak_t &remote_section_ref)
+                           const nixl_remote_section_weak_t &remote_section_ref,
+                           nixl::trace::TraceContext trace_context)
     : initiatorDescs(local_type),
```

Cppcheck (2.21.0)

[style] The function 'lock_shared' is never used.

(unusedFunction)